modules/phase_field/src/postprocessors/GrainTracker.C

```diff
@@ -836,49 +828,46 @@ GrainTracker::updateFieldInfo()
   std::map<unsigned int, Real> tmp_map;
   MeshBase & mesh = _mesh.getMesh();
 
-  for (std::map<unsigned int, MooseSharedPointer<FeatureData> >::iterator grain_it = _unique_grains.begin(); grain_it != _unique_grains.end(); ++grain_it)
+  for (const auto & grain_kv : _unique_grains)
   {
-    unsigned int curr_var = grain_it->second->_var_idx;
+    unsigned int curr_var = grain_kv.second->_var_idx;
```

Comment:
I've gone to doing something similar... but I typically use the word `pair`... i.e. I would call this `grain_pair`.  It is nice and literal sounding (that's exactly what it really is) and it helps me remember that I need to use `.first` and `.second` to grab values from it.